src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java

```diff
@@ -917,15 +921,21 @@ private PublicHeaders computePublicHeaders() {
       return new PublicHeaders(
           ImmutableList.copyOf(Iterables.concat(publicHeaders, nonModuleMapHeaders)),
           ImmutableList.copyOf(publicHeaders),
-          null);
+          /*virtualIncludePath=*/ null,
+          /* virtualToOriginalHeaders= */ NestedSetBuilder.create(Order.STABLE_ORDER));
     }
 
     if (ruleContext.hasErrors()) {
-      return new PublicHeaders(ImmutableList.of(), ImmutableList.of(), null);
+      return new PublicHeaders(
+          ImmutableList.of(),
+          ImmutableList.of(),
+          null,
```

Comment:
Up to you, my thinking with param comments is to add them whenever it's not clear what the comment represents from the variable name. I also put them in the constructors for EMPTY objects, as it simplifies understanding of "what goes into the constructor".

I agree that we do this only because the IDE is not helping with this already, but I still like them :)

Reply:
OK.